### PR TITLE
Task/sdk 3639/refactor geofence unit tests

### PR DESCRIPTION
**Pull request by Anush-Shand**
Java 17 doesn't support powermock. Hence geofence tests have been refactored

### CODE REVIEW

File: clevertap-geofence/build.gradle [hunk 1]
```diff
@@ -25,15 +25,14 @@ dependencies {
     testImplementation (libs.test.junit.platform.runner)
     testImplementation (libs.test.junit.jupiter.api)
 
-    testImplementation (libs.test.mockito.core)
+    testApi (libs.bundles.mockito)
```
Comment (CTLalit) [inline clevertap-geofence/build.gradle:28]: why did we change to api? the client will have to add it then? or is it pulling from core lib?

**Review by CTLalit — COMMENTED** [review]
Only one comment for mockstatic, why did we not use the kotlin lib version from MockK?

File: clevertap-geofence/build.gradle (hunk 1 above)
Comment (Anush-Shand) [reply clevertap-geofence/build.gradle]: Few unit tests were failing if testImplementation was used, now resolved

**Comment by Anush-Shand** [comment]
@CTLalit mockk wasn't used as then a lot of already working tests would have also needed refactoring. To make it uniform decided to go forward with Mockito
We should use mockk going forward

**Review by CTLalit — APPROVED** [review]

**Pull request merged**